da/celestia/celestia.go

```diff
@@ -269,8 +269,12 @@ func (c *DataAvailabilityLayerClient) RetrieveBatches(dataLayerHeight uint64) da
 		var batch pb.Batch
 		err = proto.Unmarshal(msg, &batch)
 		if err != nil {
-			c.logger.Error("failed to unmarshal batch", "daHeight", dataLayerHeight, "position", i, "error", err)
-			continue
+			return da.ResultRetrieveBatch{
```

Comment:
Anyone can write to a celestia da namespace. This means that every one who write a random blob which managed to fit into the same height as the rollapp batch, would crash the process. this is undesirable. in essence, lack of ability to unmarshal the blob doesn't mean that something went wrong.

Reply:
ok good to know. will fix

in this case, the original version had a bug as well, as we created batch array with `len(data)`. than it could be only partially filled
than we iterate over the batch array with nil values inside